src/LibraryManager/Manifest.cs

```diff
@@ -216,44 +230,58 @@ public Manifest Clone()
                 DestinationPath = destination
             };
 
-            UpdateLibraryProviderAndDestination(desiredState, DefaultProvider, DefaultDestination);
-            if (!desiredState.IsValid(out IEnumerable<IError> errors))
-            {
-                return new List<ILibraryOperationResult> { new LibraryOperationResult(desiredState, errors.ToArray()) };
-            }
+            return await _librariesValidator.ValidateLibraryAsync(desiredState, cancellationToken).ConfigureAwait(false);   
+        }
 
-            IProvider provider = _dependencies.GetProvider(desiredState.ProviderId);
-            if (provider == null)
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="libraryId"></param>
+        /// <param name="providerId"></param>
+        /// <param name="files"></param>
+        /// <param name="destination"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<IEnumerable<ILibraryOperationResult>> InstallLibraryAsync(string libraryId, string providerId, IReadOnlyList<string> files, string destination, CancellationToken cancellationToken)
```

Comment:
>InstallLibraryAsync [](start = 64, length = 19)

Should this just take a `LibraryInstallationState` as a parameter to simplify the argument list?

Reply:
We'd have to make `LibraryInstallationState` public, which we have been trying to avoid.

Reply:
I will let you decide on this one and make the change if needed.

---
In reply to: [195524231](https://github.com/aspnet/LibraryManager/pull/164#discussion_r195524231) [](ancestors = 195524231)